modules/redis/redis.c

```diff
@@ -246,58 +272,32 @@ redis_worker_insert(LogThreadedDestDriver *s, LogMessage *msg)
       return LTR_NOT_CONNECTED;
     }
 
-  log_template_format(self->key, msg, &self->template_options, LTZ_SEND,
-                      self->super.worker.instance.seq_num, NULL, self->key_str);
-
-  if (self->param1)
-    log_template_format(self->param1, msg, &self->template_options, LTZ_SEND,
-                        self->super.worker.instance.seq_num, NULL, self->param1_str);
-  if (self->param2)
-    log_template_format(self->param2, msg, &self->template_options, LTZ_SEND,
-                        self->super.worker.instance.seq_num, NULL, self->param2_str);
+  _fill_argv_from_template_list(self, msg);
 
-  argv[0] = self->command->str;
-  argvlen[0] = self->command->len;
-  argv[1] = self->key_str->str;
-  argvlen[1] = self->key_str->len;
-
-  if (self->param1)
-    {
-      argv[2] = self->param1_str->str;
-      argvlen[2] = self->param1_str->len;
-      argc++;
-    }
-
-  if (self->param2)
-    {
-      argv[3] = self->param2_str->str;
-      argvlen[3] = self->param2_str->len;
-      argc++;
-    }
+  redisReply *reply = redisCommandArgv(self->c, self->argc, (const gchar **)self->argv, self->argvlen);
 
-  reply = redisCommandArgv(self->c, argc, argv, argvlen);
+  ScratchBuffersMarker marker;
+  GString *full_command = _argv_to_string(self, &marker);
```

Comment:
[optional] you don't have to inject a marker into this function, simply set the marker outside:

```C
ScratchBufferMarker marker;
scratch_buffers_mark(&marker);
GString *full_command = _argv_to_string(self);
...
_free_argv_templated_part(self);
scratch_buffers_reclaim_marked(marker);
```

Reply:
Thanks, I think I finally understood how scratch_markers work, and I merged the usage of the marker with `_fill_argv_from_template_list()`.